packages/x-data-grid/src/hooks/features/overlays/useGridOverlays.ts

```diff
@@ -0,0 +1,47 @@
+import { useGridSelector } from '../../utils';
+import { useGridApiContext } from '../../utils/useGridApiContext';
+import { useGridRootProps } from '../../utils/useGridRootProps';
+import { gridExpandedRowCountSelector } from '../filter';
+import { gridRowCountSelector, gridRowsLoadingSelector } from '../rows';
+import { GridLoadingOverlayVariant } from '../../../components/GridLoadingOverlay';
+import { GridSlotsComponent } from '../../../models/gridSlotsComponent';
+
+export type GridOverlayType =
+  | keyof Pick<GridSlotsComponent, 'noRowsOverlay' | 'noResultsOverlay' | 'loadingOverlay'>
+  | null;
+
+/**
+ * Uses the grid state to determine which overlay to display.
+ * Returns the active overlay type and the active loading overlay variant.
+ */
+export const useGridOverlays = () => {
```

Comment:
Default to named function in top level scope

```suggestion
function useGridOverlays(
```